### DISCUSSION TITLE
Micropython-controlled lung elastometer

### DISCUSSION THREAD

**Discussion by JDRBoston** [micropython/micropython#12194] [Show and tell]
Thank you for all the advice that you have given me over the years! I use it to help build a micropython-controlled instrument that I use in my lab to measure lung elasticity while studying mechanisms of pediatric pulmonary disease. The device uses micropython to obtain syringe piston range and airway pressure data from sensors sitting on an i2c bus. It then uses that information to drive a stepper motor to control syringe movement and gas injection or withdraw from an attached lung. I've published an open-source article that describes the design, build, testing and use of the device (link). The article shows some pics of the device and its components; I put some videos in the on-line supplements that show the device in scripts and instrument in action. I hope that you find it to be interesting. - Jay

https://www.nature.com/articles/s41598-023-38310-7

**Comment by andrewleech** [comment]
That's brilliant, thanks for sharing!

**Comment by JDRBoston** [comment]
Thank you very much, Andrew!! I am glad that you appreciate my work. - Jay
________________________________

**Comment by jimmo** [comment]
Fantastic, thanks @JDRBoston !

**Comment by dpgeorge** [comment]
It's really interesting to know what MicroPython is being used for out in the wild.  Thank you for sharing!

**Comment by JDRBoston** [comment]
Thank you very much Jimmo for the kind way you dealt with my noob questions in the past. Damien, I think that you might have liked the Acknowledgement section of the paper. I have another recent research paper published at HardwareX about a MicroPython-powered device that I made for some of our clinical research - I'll upload it to this section shortly for your interest.

**Reply by mattytrentini** [reply]
To come full-circle, you might like to watch Damien's [Open-Source Communities and Ten Years of MicroPython](https://youtu.be/iEQQWpjgFd8?si=FUKjLoFRxAO5hDTf&t=75) from the recent PyCon AU for a mention of a certain _Lung Elastometer_ 😄 (mentioned at about [4:10](https://youtu.be/iEQQWpjgFd8?si=Yns784sdTWQCRjyc&t=248)).

**Reply by JDRBoston** [reply]
Thank you, mattytrentini! I am truly honored to have LungElast mentioned among the other very interesting and successful MicroPython use cases/projects!
